### PR TITLE
Refactor the views package for better extendability

### PR DESCRIPTION
**Pull request by iliabhz**
Made a base layout and made it so every "page" would have it's own package in the views package and a corresponding index templ file to "Show" the page. This is a common approach with templ projects as seen in this [project](https://github.com/emarifer/go-echo-templ-project-structure).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,4 +8,8 @@ serve:
 
 dev:
```
Comment (mhrlife) [inline Makefile]: If you want to refactor the Makefile, I suggest also using the generate command in the dev command, like:

dev: generate
   go run main.go ...

File: internal/webapp/views/layout/base_templ.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+package layout
+
+//lint:file-ignore SA4006 This context is only used if a nested component is present.
+
+import "github.com/a-h/templ"
+import "context"
+import "io"
+import "bytes"
+
+func Base(title string) templ.Component {
```
Comment (mhrlife) [inline internal/webapp/views/layout/base_templ.go]: Don't change this part please, passing just a title as the parameter is not a good practice. I will refactor this part and by using custom types make this template more dynamic.

File: internal/webapp/views/pages/lobby_page.templ [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
-
-<script>
-</script>
-}
+package pages
+
+import "kingscomp/internal/webapp/views/layout"
+
+templ LobbyPage(){
+@layout.Base("Lobby"){
```
Comment (mhrlife) [inline internal/webapp/views/pages/lobby_page.templ]: Also there is no need to pass the title to the base layout, I will refactor this part myself using custom types.

**Review by mhrlife — COMMENTED** [review]
Thank you for your contributions. I had some minor comments.

**Comment by iliabhz** [comment]
Thanks for the feedback, i have made some changes according to your comments and it should be good to go, let me know if there is anything else that needs to be changed.

**Pull request merged**